### PR TITLE
Fixes GMB nudge view track events not fired on site switch

### PR DESCRIPTION
**Pull request by Tug**
This fires the `calypso_google_my_business_stats_nudge_view` event on site switch.

### Testing Instructions
- Boot this branch locally
- Log analytics events in the console: localStorage.debug = 'calypso:analytics:*';
- Go to the stats page for a site that is eligible for Google My Business
- Ensure you see the `calypso_google_my_business_stats_nudge_view` event fired on page load
- Switch to a site that is not eligible for GMB
- Ensure you do not see the `calypso_google_my_business_stats_nudge_view` event
- Switch to another site that is eligible for GMB
- Ensure you see the `calypso_google_my_business_stats_nudge_view` event

### Reviews
- [ ] Code
- [ ] Product

### CODE REVIEW

File: client/blocks/google-my-business-stats-nudge/index.js [hunk 1]
```diff
@@ -32,11 +32,41 @@ class GoogleMyBusinessStatsNudge extends Component {
...
+
+		if ( state.siteSwitch ) {
+			return {
+				siteSwitch: false,
+			};
+		}
+	}
+
+	componentDidUpdate() {
+		if ( this.state.siteSwitch ) {
```
Comment (yurynix) [inline client/blocks/google-my-business-stats-nudge/index.js]: Why not check that directly here and drop `getDerivedStateFromProps` ?
```es6
componentDidUpdate( prevProps ) {
	if ( prevProps.siteId && this.props.siteId && this.props.siteId !=== prevProps.siteId ) {
		this.recordView();
	}
}
```

**Comment by yurynix** [comment]
The test scenario in the description works as described 👍

File: client/blocks/google-my-business-stats-nudge/index.js (hunk 1 above)
Comment (Tug) [reply client/blocks/google-my-business-stats-nudge/index.js]: Because `visible` might be `false` on site switch, we need to wait for the props to change.

Comment (yurynix) [reply client/blocks/google-my-business-stats-nudge/index.js]: I see 👍 Could you please write a comment about it on `getDerivedStateFromProps`?

**Comment by cbauerman** [comment]
Works as described & everything looks good to me! I was trying to accommodate changes in visibility as well, but I don't think that will *ever* be an actual issue. LGTM!

**Pull request merged**